### PR TITLE
Remove try/catch patterns within unit tests

### PR DESCRIPTION
**Pull request by ChrisAntaki**
- Removes try/catch patterns within unit tests, since they can hide silent failures
- Explicitly expects errors/rejects instead
- Adds eslint rule blocking try/catch within unit tests

### CODE REVIEW

**Comment by ChrisAntaki** [comment]
I'll send this for review after #2628

File: src/runtime/basic-runtime-test.js [hunk 1]
```diff
@@ -75,11 +75,7 @@ describes.realWin('installBasicRuntime', (env) => {
     expect(progress).to.equal('');
 
     // Install runtime and schedule few more dependencies.
-    try {
-      installBasicRuntime(win);
-    } catch (e) {
-      // Page doesn't have valid subscription and hence this function throws.
```
Comment (ChrisAntaki) [inline src/runtime/basic-runtime-test.js:81]: Narrator: It didn't

Comment (jpettitt) [reply src/runtime/basic-runtime-test.js:81]: LOL

Comment (jpettitt) [reply src/runtime/basic-runtime-test.js:81]: But should it have?

Comment (ChrisAntaki) [reply src/runtime/basic-runtime-test.js:81]: Haha, my best guess is no... at least from what's going on in the test and its name

**Pull request merged**